core/devmode/src/main/java/io/quarkus/dev/ClassLoaderCompiler.java

```diff
@@ -122,12 +122,19 @@ public ClassLoaderCompiler(ClassLoader classLoader,
                             }
                             Object classPath = mf.getMainAttributes().get(Attributes.Name.CLASS_PATH);
                             if (classPath != null) {
-                                for (String i : WHITESPACE_PATTERN.split(classPath.toString())) {
+                                for (String classPathEntry : WHITESPACE_PATTERN.split(classPath.toString())) {
+                                    final URI cpEntryURI = new URI(classPathEntry);
```

Comment:
Use `URI.create` here

Reply:
Hello @gastaldi, going by the javadoc of `URI.create`[1], it states that:

```
...
 The constructors, which throw URISyntaxException directly, should be used situations where a URI is being constructed from user input or from some other source that may be prone to errors. 
```
Given that this above code is parsing (random) jar files' `Class-Path` entries, I think using the constructor instead of `URI.create` would be the right thing to do.

[1] https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#create-java.lang.String-